main.rb

```diff
@@ -0,0 +1,178 @@
+# rubocop:disable Metrics/ModuleLength, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  # =========
+  #  my_each
+  # =========
+
+  def my_each(_arr = nil)
+    return to_enum unless block_given?
+
+    # variable for the array's length
+    arr_size = Array(self).length
+    # loops and returns each number in the array
+    arr_size.times do |i|
+      yield(Array(self)[i])
+    end
+    self
+  end
+
+  # ====================
+  #  my_each_with_index
+  # ====================
+
+  def my_each_with_index(_arr = nil)
+    return to_enum unless block_given?
+
+    arr_size = Array(self).length
+    # loops and returns each number and index in the array
+    arr_size.times do |i|
+      yield(Array(self)[i], i)
+    end
+    self
+  end
+
+  # ===========
+  #  my_select
+  # ===========
+
+  def my_select(_arr = nil)
+    return to_enum unless block_given?
+
+    # new array that contains the passed arguments
+    new_arr = []
+    arr_size = Array(self).length
+    # loop through array and pushes elements that are true
+    arr_size.times do |x|
+      new_arr.push(Array(self)[x]) if yield(Array(self)[x])
+    end
+    new_arr
+  end
+
+  # ========
+  #  my_all
+  # ========
+
+  def my_all?(arg = nil)
+    result = false
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result unless i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result unless x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result unless y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result if a.nil? }
+    elsif block_given?
+      my_each { |b| return result unless yield(b) }
+    end
+    !result
+  end
+
+  # ========
+  #  my_any
+  # ========
+
+  def my_any?(arg = nil)
+    result = true
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
```

Comment:
- [ ] If no block or argument is given, `my_any` should return false if at least one of the collection is not true

```ruby 
 [ nil, false, false ].any? == [ nil, false, false ].my_any? => false
```